bin/heudiconv

```diff
@@ -209,38 +308,44 @@ def conversion_info(subject, outdir, info, filegroup, basedir=None):
     return convert_info
 
 
-def embed_nifti(dcmfiles, niftifile, infofile, force=False):
-    import dcmstack as ds
-    import nibabel as nb
-    import os
-    stack = ds.parse_and_stack(dcmfiles, force=force).values()
-    if len(stack) > 1:
-        raise ValueError('Found multiple series')
-    stack = stack[0]
-
-    #Create the nifti image using the data array
-    if not os.path.exists(niftifile):
-        nifti_image = stack.to_nifti(embed_meta=True)
-        nifti_image.to_filename(niftifile)
-        return ds.NiftiWrapper(nifti_image).meta_ext.to_json()
-    orig_nii = nb.load(niftifile)
-    #orig_hdr = orig_nii.get_header()
-    aff = orig_nii.get_affine()
-    ornt = nb.orientations.io_orientation(aff)
-    axcodes = nb.orientations.ornt2axcodes(ornt)
-    new_nii = stack.to_nifti(voxel_order=''.join(axcodes), embed_meta=True)
-    #new_hdr = new_nii.get_header()
-    #orig_hdr.extensions = new_hdr.extensions
-    #orig_nii.update_header()
-    #orig_nii.to_filename(niftifile)
-    meta = ds.NiftiWrapper(new_nii).meta_ext.to_json()
-    with open(infofile, 'wt') as fp:
-        fp.writelines(meta)
+def embed_nifti(dcmfiles, niftifile, infofile, force=False, infofile_style='dcmstack'):
+    if infofile_style == 'dcmstack':
+        import dcmstack as ds
+        import nibabel as nb
+        import os
+        stack = ds.parse_and_stack(dcmfiles, force=force).values()
+        if len(stack) > 1:
+            raise ValueError('Found multiple series')
+        stack = stack[0]
+    
+        #Create the nifti image using the data array
+        if not os.path.exists(niftifile):
+            nifti_image = stack.to_nifti(embed_meta=True)
+            nifti_image.to_filename(niftifile)
+            return ds.NiftiWrapper(nifti_image).meta_ext.to_json()
+        orig_nii = nb.load(niftifile)
+        #orig_hdr = orig_nii.get_header()
+        aff = orig_nii.get_affine()
+        ornt = nb.orientations.io_orientation(aff)
+        axcodes = nb.orientations.ornt2axcodes(ornt)
+        new_nii = stack.to_nifti(voxel_order=''.join(axcodes), embed_meta=True)
+        #new_hdr = new_nii.get_header()
+        #orig_hdr.extensions = new_hdr.extensions
+        #orig_nii.update_header()
+        #orig_nii.to_filename(niftifile)
+        meta = ds.NiftiWrapper(new_nii).meta_ext.to_json()
+    
+        with open(infofile, 'wt') as fp:
+            fp.writelines(meta)
+    elif infofile_style == 'bids':
+        # we are not taking the first file because some multiband sequences report wrong slice timing in the first volume
+        extract_metadata(dcmfiles[-1], infofile)
```

Comment:
the other issue with this statement is that it assumes that all files have consistent info. something dcmstack clearly separates into static and temporal info. i think that should be maintained.